### PR TITLE
YouTube search. closes #9, closes #73

### LINKED ISSUE #9: YouTube search
Just need to retrieve results from the API, and hide any images - then make the links do some cool redirection to Zen Audio Player by passing the YouTube video URL as a querystring parameter.

### LINKED ISSUE #73: Invalid video ID causes odd behavior
## What happens:

1. We get the video description error
2. Some elements "reload"
3. We get the correct error
4. The :musical_note: icon shows up where the title should be (but it's an invalid video ID so there is no title)
5. The time shows up!

capturfiles_7

## What should happen:

1. We get the correct error

## Repro steps:

1. Navigate to https://zen-audio-player.github.io/?v=test
2. Watch in amazement :astonished:

### PR DESCRIPTION
**Pull request by monicacheung**
closes #9, closes #73

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -46,11 +46,16 @@
                 </span>
             </div>
         </form>
+        <div id="search-results">
+            <h3>Search Results</h3>
+            <ul></ul>
+        </div>
+        <br>
```
Comment (shakeelmohamed) [inline index.html]: Let's remove this one

Comment (monicacheung) [inline index.html]: What are you referring to?

Comment (shakeelmohamed) [inline index.html]: The `<br>`

File: js/everything.js [hunk 2]
```diff
@@ -70,6 +68,222 @@ function onYouTubeIframeAPIReady() { //eslint-disable-line no-unused-vars
...
+};
+
+function isFileProtocol() {
+    return window.location.protocol === "file:";
+}
+
+// Lock for updating the volume
+var VOLUME_LOCKED = false;
+
+var zenPlayer = {
```
Comment (shakeelmohamed) [inline js/everything.js]: nit: since this is basically a class, how about `ZenPlayer`?

File: js/everything.js [hunk 3]
```diff
@@ -70,6 +68,222 @@ function onYouTubeIframeAPIReady() { //eslint-disable-line no-unused-vars
...
+        $("#zen-video-title").attr("href", this.videoUrl);
+    },
+    setupVideoDescription: function() {
+        var description = anchorURLs(this.videoDescription);
+        $("#zen-video-description").html(description);
+        $("#zen-video-description").hide();
+
+        $("#toggleDescription").click(function(event) {
+            event.preventDefault();
+            var descriptionElement = $("#zen-video-description");
```
Comment (shakeelmohamed) [inline js/everything.js]: nit: blank line after `event.preventDefault();`

**Comment by shakeelmohamed** [comment]
@monicacheung Let me just say,

# This is awesome work!

Thank you for adding one one of the biggest features!
I've left a few nits, overall this set of changes looks fantastic.

Can you apply the following patch, we should be using strict equality at all times:

```diff
diff --git a/.eslintrc.json b/.eslintrc.json
index f9afc98..9315d7f 100644
--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -44,7 +44,8 @@
         "semi": [
             2,
             "always"
-        ]
+        ],
+        "eqeqeq": 2
     },
     "extends": "eslint:recommended"
 }
diff --git a/js/everything.js b/js/everything.js
index ba307c4..9d5fd6b 100644
--- a/js/everything.js
+++ b/js/everything.js
@@ -74,7 +74,7 @@ function onPlayerReady(event) {
     updateTweetMessage();
 
     // If the video isn't going to play, then return.
-    if (event.target.getPlayerState() != YT.PlayerState.BUFFERING) {
+    if (event.target.getPlayerState() !== YT.PlayerState.BUFFERING) {
         if (currentVideoID.length > 0) {
             errorMessage.show("Invalid YouTube videoID or URL.");
         }
```

**Comment by monicacheung** [comment]
Committed the changes. Glad you are happy! This was great fun. :smile:

**Pull request merged**